smarc_bt/src/mission_plan.py

```diff
@@ -473,6 +539,200 @@ def path_to_list(self, path_msg):
         return wps
 
 
+    def circle_line_segment_intersection(self, circle_center, circle_radius, pt1, pt2, full_line=False, tangent_tol=1e-5):
```

Comment:
Both this and the dubins_mission_planner methods should really be in dubins.py, I'd rather keep the mission plan for wp-selection and wp-creation logic only. Similar to how coverage planner is its own thing.